src/main/java/ch/njol/skript/effects/EffBroadcast.java

```diff
@@ -76,33 +70,17 @@ public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelaye
 	}
 	
 	@Override
-	@SuppressWarnings("deprecation")
-	public void execute(Event e) {
+	public void execute(Event event) {
 		List<CommandSender> receivers = new ArrayList<>();
 		if (worlds == null) {
 			receivers.addAll(Bukkit.getOnlinePlayers());
 			receivers.add(Bukkit.getConsoleSender());
 		} else {
-			for (World world : worlds.getArray(e))
+			for (World world : worlds.getArray(event))
 				receivers.addAll(world.getPlayers());
 		}
 
-		for (Expression<?> message : getMessages()) {
-			if (message instanceof VariableString) {
-				BaseComponent[] components = BungeeConverter.convert(((VariableString) message).getMessageComponents(e));
-				receivers.forEach(receiver -> receiver.spigot().sendMessage(components));
-			} else if (message instanceof ExprColoured && ((ExprColoured) message).isUnsafeFormat()) { // Manually marked as trusted
-				for (Object realMessage : message.getArray(e)) {
-					BaseComponent[] components = BungeeConverter.convert(ChatMessages.parse((String) realMessage));
-					receivers.forEach(receiver -> receiver.spigot().sendMessage(components));
-				}
-			} else {
-				for (Object messageObject : message.getArray(e)) {
-					String realMessage = messageObject instanceof String ? (String) messageObject : Classes.toString(messageObject);
-					receivers.forEach(receiver -> receiver.sendMessage(realMessage));
-				}
-			}
-		}
+		EffMessage.sendMessage(event, getMessages(), receivers.toArray(new CommandSender[0]), null);
```

Comment:
Actually if you're converting it to an array here you could just make it an array in the first place since you know the capacity (online) already.